src/node_buffer.cc

```diff
@@ -835,6 +884,108 @@ void WriteDoubleBE(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+template <typename T, enum Endianness endianness,
+         T (*strtoT)(const char*, char**, int), T min, T max>
+void WriteInt64Generic(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  CHECK(args[1]->IsString() || args[1]->IsNumber());
+  bool should_assert = args.Length() < 4;
+
+  if (should_assert) {
+    THROW_AND_RETURN_UNLESS_BUFFER(env, args[0]);
+  }
+
+  Local<ArrayBufferView> ts_obj = args[0].As<ArrayBufferView>();
+  ArrayBuffer::Contents ts_obj_c = ts_obj->Buffer()->GetContents();
+  const size_t ts_obj_offset = ts_obj->ByteOffset();
+  const size_t ts_obj_length = ts_obj->ByteLength();
+  char* const ts_obj_data =
+    static_cast<char*>(ts_obj_c.Data()) + ts_obj_offset;
+  if (ts_obj_length > 0)
+    CHECK_NE(ts_obj_data, nullptr);
+
+  T val;
+  if (args[1]->IsNumber()) {
+    val = args[1]->IntegerValue();
```

Comment:
ditto (using the non-deprecated overload taking a `context` argument)

Reply:
I didn't realize that this overload is deprecated. Is this documented somewhere?

Reply:
@seishun Yes:

https://github.com/nodejs/node/blob/94be2b17938d5ecdf1d40317d0bdee46906bfe4b/deps/v8/include/v8.h#L2297-L2309

And also [V8 API changes](https://docs.google.com/document/d/1g8JFi8T_oAE_7uAri7Njtig7fKaPDfotU6huOa1alds/edit):

> ## Ongoing and Planned Changes
> 
> ... APIs that are marked as `V8_DEPRECATE_SOON` will be marked as `V8_DEPRECATED` in the future. APIs marked as `V8_DEPRECATED` will usually be removed after one V8 release.
>
> - Introduction of MaybeLocal<> and Maybe<> APIs: [bug](https://code.google.com/p/v8/issues/detail?id=3929)
> - ...